Userland/Services/LookupServer/DNSQuestion.h

```diff
@@ -11,23 +11,29 @@
 
 namespace LookupServer {
 
+#define MDNS_UNICAST_QUESTION 0x8000
+
 class DNSQuestion {
 public:
-    DNSQuestion(const DNSName& name, u16 record_type, u16 class_code)
+    DNSQuestion(const DNSName& name, u16 record_type, u16 class_code, bool mdns_unicast_question)
         : m_name(name)
         , m_record_type(record_type)
         , m_class_code(class_code)
+        , m_mdns_unicast_question(mdns_unicast_question)
     {
     }
 
     u16 record_type() const { return m_record_type; }
     u16 class_code() const { return m_class_code; }
+    u16 raw_class_code() const { return (u16)m_class_code | (m_mdns_unicast_question ? MDNS_UNICAST_QUESTION : 0); }
     const DNSName& name() const { return m_name; }
+    bool mdns_unicast_question() const { return m_mdns_unicast_question; }
 
 private:
     DNSName m_name;
     u16 m_record_type { 0 };
     u16 m_class_code { 0 };
+    bool m_mdns_unicast_question { false };
```

Comment:
Nitpick: can we go with a more descriptive name? Perhaps `m_mdns_unicast_response_accepted`?

(In systemd-resolved, I went for `DNS_QUESTION_WANTS_UNICAST_REPLY`)

Reply:
Sounds good. I've renamed this to `m_mdns_wants_unicast_response`. How about that?